From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Sun, 27 Sep 2020 11:04:55 +0100
Subject: DESTDIR

Make go.sh listen to DESTDIR.

Set option so go.sh will terminate on an intermediate error.
---
 go.sh | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/go.sh b/go.sh
index 3fa7957..23b859d 100755
--- a/go.sh
+++ b/go.sh
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
+
 # This script installs the firmware and topologies in /lib/firmware/intel.
 # Optionally if ROOT is set as an environment variable by the caller it will be
 # prepended to the installation path.
@@ -23,16 +25,17 @@ test -d ${INTEL_PATH}/sof-tplg-${SOF_VERSION} || \
 echo "Installing Intel firmware and topology $SOF_VERSION to $INTEL_PATH"
 
 # wipe previous releases
-rm -rf ${ROOT}/${INTEL_PATH}/sof/*
-rm -rf ${ROOT}/${INTEL_PATH}/sof-tplg-*
-rm -rf ${ROOT}/${INTEL_PATH}/sof-tplg
+rm -rf ${DESTDIR}${ROOT}/${INTEL_PATH}/sof/*
+rm -rf ${DESTDIR}${ROOT}/${INTEL_PATH}/sof-tplg-*
+rm -rf ${DESTDIR}${ROOT}/${INTEL_PATH}/sof-tplg
 
 # copy to destination
 cd lib/firmware
-cp -rf intel ${ROOT}/lib/firmware
+mkdir -p ${DESTDIR}${ROOT}/lib/firmware
+cp -rf intel ${DESTDIR}${ROOT}/lib/firmware
 
 # add symlinks
-cd ${ROOT}/${INTEL_PATH}/sof
+cd ${DESTDIR}${ROOT}/${INTEL_PATH}/sof
 
 ln -s ${SOF_VERSION}/sof-bdw-${SOF_VERSION}.ri sof-bdw.ri
 ln -s ${SOF_VERSION}/sof-byt-${SOF_VERSION}.ri sof-byt.ri
@@ -47,4 +50,4 @@ ln -s ${SOF_VERSION}/intel-signed/sof-icl-${SOF_VERSION}.ri sof-icl.ri
 cd ..
 ln -s sof-tplg-${SOF_VERSION} sof-tplg
 
-echo "Done installing Intel firmware and topology $SOF_VERSION"
\ No newline at end of file
+echo "Done installing Intel firmware and topology $SOF_VERSION"
